### DISCUSSION TITLE
Jyers fork of which Marlin? And Arc Welder?

### DISCUSSION THREAD

**Discussion by mostlyflawles** [Jyers/Marlin#943] [General Discussion]
Hi,
Excuse me if this is a stupid question, I don't completely understand  this with branches and/or forks.
As I understand Jyers is a fork (based on) of the Marlin firmware, but which Marlin version is the latest Jyers forked from?

And a indirect related question, does Jyers firmware support Arc Welder?

best regards M

**Comment by Jyers** [comment]
This is a fork of the Bugfix 2.0.x branch of marlin, so its about as up to date as you can get. And yes, it supports arc welder!

**Reply by mostlyflawles** [reply]
Oki, thanks.

**Reply by casterle** [reply]
> This is a fork of the Bugfix 2.0.x branch of marlin, so its about as up to date as you can get. And yes, it supports arc welder!

So if this is a fork of the bugfix branch, does the fork somehow inherit changes from that branch? I'm not sure how that could work without creating conflicts, but I'm unfamiliar with the workings of GitHub. If not, how do fixes and improvements to Marlin make their way to your fork?

**Reply by Jyers** [reply]
Being a fork of the bugfix branch means it is not only based of that code, but that the git repo is linked the upstream bugfix branch. Changes to the bugfix branch are not immediately brought to this branch, but come over when I perform a pull. When I pull from the main repo, I merge any changed made upstream since my last pull as well as handling any merge conflicts that exist due to my changes. This means the branch is very up to date as it get's pulled fairly often or at least whenever important changes are made.

**Comment by casterle** [comment]
Thanks for the explanation. It’s great that GitHub supports syncing with the parent branch. How much trouble has it been fixing conflicts, especially given the major refactoring that was/is underway? Anyway, thanks for all the effort you’ve put into this!

 

-Leroy

**Reply by Jyers** [reply]
It's usually not too bad. For the most part it's just configs that give some conflicts. All the dwin code is new, so it just ignores any changes they make to the old stuff.
